### PR TITLE
Initialize XGBoost codegen

### PR DESCRIPTION
**Pull request by Yancey0623**
Initliaze XGBoost codegen, this PR reuse `db_generator` to generate XGBoost DMatrix object, part work of #749 .

### CODE REVIEW

**Review by sperlingxx — DISMISSED** [review]
LGTM

File: sql/codegen_ant_xgboost.go [hunk 1]
```diff
@@ -795,7 +795,7 @@ func xgCreatePredictionTable(pr *extendedSelect, r *antXGBoostFiller, db *DB) er
 	return nil
 }
 
-func genXG(w io.Writer, pr *extendedSelect, ds *trainAndValDataset, fts fieldTypes, db *DB) error {
+func genAntXGboost(w io.Writer, pr *extendedSelect, ds *trainAndValDataset, fts fieldTypes, db *DB) error {
```
Comment (weiguoz) [inline sql/codegen_ant_xgboost.go]: how about renaming it to genAntXG**B**oost?

Comment (Yancey0623) [reply sql/codegen_ant_xgboost.go]: You're right, that's a typo, thx.

File: sql/codegen_xgboost.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+	return r, nil
+}
+
+func genXGBoost(w io.Writer, pr *extendedSelect, ds *trainAndValDataset, fts fieldTypes, db *DB) error {
+	r, e := newXGBFiller(pr, ds, fts, db)
+	if e != nil {
+		return e
+	}
+	if pr.train {
+		fmt.Println(r.TrainCfgJSON)
```
Comment (typhoonzero) [inline sql/codegen_xgboost.go]: Is this for debug? if not please use `log.Infof`

File: sql/codegen_xgboost.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+	Maximize      bool `json:"maximize,omitempty"`
+}
+
+type xgbFiller struct {
+	IsTrain              bool
+	TrainingDatasetSQL   string
+	ValidationDatasetSQL string
+	TrainCfg             *xgbTrainConfig
+	Features             []*featureMeta
+	Label                *featureMeta
```
Comment (weiguoz) [inline sql/codegen_xgboost.go:33]: I found our [filler for tensorflow](https://github.com/sql-machine-learning/sqlflow/blob/develop/sql/codegen.go#L62) in `codegen.go`,
https://github.com/sql-machine-learning/sqlflow/blob/9a0dc86b160c4248732124e92ea64edcfc72a41f/sql/codegen.go#L67-L69
`Features` named to `X` and `Label` to `Y`.
I would suggest such consistent naming.

File: sql/expression_resolver_xgb.go [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+	"strconv"
+)
+
+type resolvedXGBTrainClause struct {
+	NumBoostRound int
+	Maximize      bool
+	ParamsAttr    map[string]*attribute
+}
+
+func resolveXGBTrainClause(tc *trainClause) (*resolvedXGBTrainClause, error) {
```
Comment (typhoonzero) [inline sql/expression_resolver_xgb.go]: try put attribute extraction into the codegen for each submitter?

File: sql/codegen_xgboost.go (hunk 2 above)
Comment (weiguoz) [inline sql/codegen_xgboost.go]: Remove this line.

File: sql/expression_resolver_xgb.go (hunk 4 above)
Comment (Yancey0623) [reply sql/expression_resolver_xgb.go]: It's used to resolve XGBoost parameters, the current `expression_resover` can only resolve the Tensorlfow parameters, maybe we can refactor the `expression_resolver` to extract `getBoolAttr`, `getIntAttr` as the common function.

File: sql/codegen_xgboost.go (hunk 2 above)
Comment (Yancey0623) [reply sql/codegen_xgboost.go]: done.

Comment (Yancey0623) [reply sql/codegen_xgboost.go]: done.

File: sql/codegen_xgboost.go (hunk 3 above)
Comment (Yancey0623) [reply sql/codegen_xgboost.go:33]: Maybe a better way is reuse the filler in `codegen`, would optimize it in the next PR.

**Review by weiguoz — DISMISSED** [review]
LGTM

File: sql/expression_resolver_xgb.go (hunk 4 above)
Comment (typhoonzero) [reply sql/expression_resolver_xgb.go]: If we only have these two parameters currently, try all put it in `expression_resolver.go` and refine this later. We should let each codegen to deal with it's only attributes.

**Review by typhoonzero — CHANGES_REQUESTED** [review]

**Comment by Yancey0623** [comment]
> If we only have these two parameters currently, try all put it in expression_resolver.go and refine this later. We should let each codegen to deal with it's only attributes.

@typhoonzero , there is not only two parameters, and I removed the xgb resolver from this PR, would refactor the `expression_resolver.go` in the next PR.

**Review by typhoonzero — APPROVED** [review]
